File: signature/vuejs.go
package signature

var VueJsSignature = Signature{
	Name:        "Vue.js",
	Description: "Vue.js is a progressive JavaScript framework for building user interfaces.",
	Cpe:         "cpe:/a:vuejs:vuejs",

	DetectPattern: DetectPattern{
		URLs: []string{
			"/vue.js",
			"/vue.min.js",
		},
		Bodies: []string{
			"Vue.js",
		},
	},
	VersionPattern: VersionPattern{
		Bodies: []string{
			"Vue.js v(\\d+\\.\\d+\\.\\d+)",
			"version=\"(\\d+\\.\\d+\\.\\d+)\"",
			"vue v(\\d+\\.\\d+\\.\\d+)",
		},
	},
}

File: src/signatures/technologies/vue_js.ts
import type { Signature } from "../_types.js";

export const vueJsSignature: Signature = {
  name: "Vue.js",
  description:
    "Vue.js is an open-source model-view-viewmodel JavaScript framework for building user interfaces and single-page applications.",
  rule: {
    confidence: "high",
    urls: [
      "vue[.-]([\\d.]*\\d)[^/]*\\.js",
      "(?:/([\\d.]+))?/vue(?:\\.min)?\\.js",
    ],
    bodies: ["<[^>]+\\sdata-v(?:ue)?-", "vue-app"],
    javascriptVariables: {
      Vue: "",
      "Vue.version": "(.+)",
      VueRoot: "",
      __VUE_HOT_MAP__: "",
      __VUE__: "",
      vueDLL: "",
    },
  },
};
